[raihahahan] iP

raihahahan opened the pull request:
<div id="top"></div>

<!-- PROJECT LOGO -->
<br />
<div align="center">
    <img src="https://cdn-icons-png.flaticon.com/512/4712/4712139.png" alt="Logo" width="80" height="80">

  <h3 align="center">Susan Bot</h3>

  <p align="center">
    Susan bot helps you manage tasks well.
    <br />
    <a href="https://nus-cs2103-ay2425s1.github.io/website/schedule/week3/project.html"><strong>Explore the docs »</strong></a>
    <br />
    <br />
  </p>
</div>

<!-- TABLE OF CONTENTS -->
<details>
  <summary>Table of Contents</summary>
  <ol>
    <li>
      <a href="#about-the-project">About The Project</a>
    </li>
    <li>
      <a href="#features">Features</a>
    </li>
    <li>
      <a href="#getting-started">Getting Started</a>
    </li>
    <li><a href="#roadmap">Roadmap</a></li>
  </ol>
</details>

<!-- ABOUT THE PROJECT -->

## About The Project

Susan Bot is a simple lightweight chat bot that lets you manage tasks easily from your desktop. In this PR, Susan Bot is a CLI program.

## Features:

1. `todo <description>`: Adds tasks without any date/time attached to it e.g., visit new theme park
2. `deadline <description> /by <deadline>`: adds tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
3. `event <description> /from <from> /to <to>`: adds tasks that start at a specific date/time and ends at a specific date/time e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019
4. `list`: Lists down your current tasks.
5. `mark <taskNumber>`: Checks the task with taskNumber as done.
6. `unmark <taskNumber>`: Checks the task with taskNumber as undone.
7. `bye`: Exits the program.
8. `delete <taskNumber>`: Deletes the task with taskNumber.

<p align="right">(<a href="#top">back to top</a>)</p>

<!-- GETTING STARTED -->

## Getting Started

1. Set up tools: https://nus-cs2103-ay2425s1.github.io/website/schedule/week1/admin.html
2. `git clone https://github.com/raihahahan/ip.git`
3. Gradle build
4. Run ip [run] with Gradle.

<p align="right">(<a href="#top">back to top</a>)</p>

<!-- ROADMAP -->

## Roadmap

- [x] Week 1 Tasks
- [x] Week 2 Tasks
- [x] Week 3 Tasks
- [x] Week 4 Tasks

<p align="right">(<a href="#top">back to top</a>)</p>

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

raihahahan committed:
Level-0 initial commit

raihahahan committed:
Fixed tagging issue

raihahahan committed:
added meowing

raihahahan committed:
Level-2

raihahahan committed:
Heavy refactoring for modularity

raihahahan committed:
Level-4: ToDo, Event, Deadline

raihahahan committed:
A-TextUiTesting

raihahahan committed:
Level-5: Handle errors

raihahahan committed:
Level-6: Delete

raihahahan committed:
Added enums

raihahahan committed:
made many refactors. added save feature and packaged classes

raihahahan committed:
Merge pull request #1 from raihahahan/branch-Level-7

made many refactors. added save feature and packaged classes

raihahahan committed:
added Date handling

raihahahan committed:
Merge pull request #2 from raihahahan/branch-Level-8

added Date handling

raihahahan committed:
added more classes

raihahahan committed:
already divided into packages

raihahahan committed:
Merge pull request #3 from raihahahan/add-gradle-support

Add gradle support

raihahahan committed:
added gradle support

raihahahan committed:
added junit

raihahahan committed:
jar file creation

raihahahan committed:
added javadoc

raihahahan committed:
Merge pull request #4 from raihahahan/branch-A-JavaDoc

added javadoc

raihahahan committed:
fixed code style

raihahahan committed:
Merge pull request #5 from raihahahan/branch-A-CodingStandard

fixed code style

raihahahan committed:
created find feature

raihahahan committed:
Merge pull request #6 from raihahahan/branch-Level-9

created find feature

nws321 commented on src/main/java/Susan.java:
```diff
@@ -0,0 +1,50 @@
+import features.ui.Ui;
+import utils.Utils;
+
+import java.util.*;
```
Advised to import specific files requires only instead of all using *

nws321 commented on src/main/java/data/TaskDAO.java:
```diff
@@ -0,0 +1,14 @@
+import java.util.Optional;
+import features.task.Task;
+
+public interface TaskDAO {
```
Slightly ambiguous what TaskDao means, suggest to have code comments about the interface

nws321 commented on src/main/java/features/command/CommandHandler.java:
```diff
@@ -0,0 +1,247 @@
+import features.task.Task;
+import utils.Utils;
+import config.Config;
+import java.util.*;
```
...similar issue as above

nws321 commented on src/main/java/features/task/TaskManagement.java:
```diff
@@ -0,0 +1,123 @@
+	 *
+	 * @param task the task to be added
+	 */
+	public void add(Task task) {
```
could indicate what is being added or removed in the naming

nws321 reviewed (commented):
I think overall the code is well packaged and intuitive, a lot of testing can be seen which is great.

ritvi12 commented on src/main/java/Susan.java:
Should list all imports explicitly.

ritvi12 commented on src/main/java/data/CSVTaskDAO.java:
Could have better class names. Not really sure what CSVTaskDAO means.

ritvi12 commented on src/main/java/data/CSVTaskDAO.java:
```diff
@@ -0,0 +1,140 @@
+            String extraData = fields.length > 4 ? fields[4].trim() : "";
+            Task task = null;
+
+            switch (type) {
```
There should be no indentation for case clauses in switch case.

ritvi12 commented on src/main/java/features/command/CommandHandler.java:
```diff
@@ -0,0 +1,247 @@
+	 */
+	public void handleCommand(Command cmd) {
+		String command = cmd.getName();
+		try {
```
Can check the indentation style for  try-catch statement from Java coding standard doc.

ritvi12 commented on src/main/java/features/task/EventTask.java:
```diff
@@ -0,0 +1,60 @@
+public class EventTask extends Task {
+	private LocalDateTime startAt;
+	private LocalDateTime endAt;
+	private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
can make this a constant.

ritvi12 commented on src/test/java/features/task/CommandHandlerTest.java:
```diff
@@ -0,0 +1,51 @@
+package features.task;
+
+import static org.junit.jupiter.api.Assertions.*;
```
should list all imports explicitly

ritvi12 reviewed (commented):
Overall, it looks good! 
Just need some tweaks in indentation of the code as well as java docs for a few classes to help better understand. Should also mention all imports explicitly in all classes.

raihahahan committed:
Implemented Level-10

raihahahan committed:
Merge pull request #7 from raihahahan/branch-Level-10

Implemented Level-10

raihahahan committed:
Update README.md

raihahahan committed:
Update README.md

raihahahan committed:
Add asserts for certain parts of the codebase

raihahahan committed:
Improve code quality of some files

raihahahan committed:
Improve code quality of some files

Variables that do not change are set as final. However, in some cases, variables are not set to final as they may be changed in future iterations.

StringBuilder uses chained append() calls instead of relying on the "+" operator.

Removed unused variables.

raihahahan committed:
Convert for loops to streams

Some areas of the code can be simply replaced from a for loop to a simple chained stream

raihahahan committed:
Merge pull request #8 from raihahahan/branch-A-Assertions

Add asserts for certain parts of the codebase

raihahahan committed:
Merge pull request #9 from raihahahan/branch-A-CodeQuality

Branch a code quality

raihahahan committed:
Merge pull request #10 from raihahahan/branch-A-Streams

Convert for loops to streams

raihahahan committed:
Add github workflow for push and pull request

raihahahan committed:
Add deleteMany command to do mass delete

raihahahan committed:
Merge pull request #11 from raihahahan/branch-C-MassOps

Add deleteMany command to do mass delete

raihahahan committed:
Add styles.css to style the GUI

raihahahan committed:
Merge pull request #13 from raihahahan/branch-A-BetterGui

Add styles.css to style the GUI

raihahahan committed:
Add personality by handling greeting in multiple languages.

raihahahan committed:
Merge pull request #14 from raihahahan/branch-A-Personality

Add personality by handling greeting in multiple languages.

raihahahan committed:
Add custom exceptions.

raihahahan committed:
Merge pull request #15 from raihahahan/branch-A-MoreErrorHandling

Add custom exceptions.

raihahahan committed:
Update readme

raihahahan committed:
Update readme and fixed build failure

raihahahan committed:
Update README.md

raihahahan committed:
A-UserGuide

raihahahan committed:
Fixed errors and cleaned code

raihahahan committed:
Added file creation if does not exists.

raihahahan committed:
Add comment for regex expression

raihahahan committed:
Rename Ui file

raihahahan committed:
Add Javadocs for some classes

raihahahan committed:
Remove wildcard imports

raihahahan committed:
Remove unnecessary whitespaces

raihahahan committed:
Fix code warnings

raihahahan committed:
Update README.md

raihahahan committed:
Add smoke tests task in README.md

raihahahan committed:
Fix JavaFX problems
